hatchways/team-oak
# Stripe API setup, Stripe Connect setup

Connected to the following issues:
https://github.com/hatchways/team-oak/issues/40
https://github.com/hatchways/team-oak/issues/63

### What this PR does (required):
- Stripe libraries added.
- Set up a new Stripe account for the team. Credentials and API keys will be shared separately.
- Sample environment variables updated with placeholders for API keys, as well as redirect links used by Stripe.
- Settings > Billing currently includes a button for petsitters. When clicked, it will redirect them to setup their account on Stripe. Redirects back to our site when done.
- Profile model updated so that we store the user's Stripe ID.
- Stripe controller  and router updated on the backend as well.

### Screenshot:
![Screen Shot 2022-01-21 at 12 15 19 AM](https://user-images.githubusercontent.com/96796802/150470103-87e1d7d1-6652-4b42-b487-f4ee1595ba22.png)

### Any information needed to test this feature:
- Sign in, go to settings > billing, click the `Connect to Stripe` button. For now, this Button is disabled after a stripe account is created on the first click (even if they haven't fully onboarded yet). 
- Complete the onboarding process as a test user (credit card and ID information can be skipped as a test user)
- Stripe provides a link to return to our site - test this too.
- Log into team oak's stripe account (test mode) to see list of users connected, and whether they have been onboarded. 

### Any issues with the current functionality (optional):
- Would be nice to preload some more information before generating stripe connect link. Stripe is strict about the data they accept, and our demo address and phone number aren't currently accepted. 
- Our profile model only tracks whether a user has a stripe ID. This does not necessarily mean that the user has completed onboarding through stripe - further work to be done on checking onboarding status, and FE work to prompt user to complete the process.
- Stripe link expires at a certain time (when we get the redirect url, the Stripe API also returns the expiry time). Should prompt petsitter to create a new link if they haven’t finished onboarding.

## Notes for the Future:
- If we ever push the app to production - remember to change the refresh and return links provided to Stripe. These are saved to the .env variable so we'll be more likely to remember.
